### DISCUSSION TITLE
The question about RainbowKitProvider creating an empty div

### DISCUSSION THREAD

**Discussion by fengcms** [rainbow-me/rainbowkit#1840] [Ideas]
When I use the RainbowKitProvider provider, an empty div with the [data-rk] attribute is created under the div#root of the page. This makes me uncomfortable because in my opinion, this is an unnecessary level.  Can we remove this empty div?

If this is an essential design in complex applications, then in most simple projects, I think it can be removed. For example, adding an option to bind the data-rk attribute to div#root or other specified className or id.

I hope you can adopt my suggestion.  I have a little obsessive-compulsive disorder.

In addition, I am not a native English speaker, and this text was translated by a translation software.  If it is inaccurate, please forgive me.

**Comment by 0xmkh** [comment]
@fengcms Thanks for creating this discussion.

I see the concern you have, but this is part of our design system. Yes we could use `className`, `id` or add it to `#root`, but that would complicate our design system even more. We've had no users complaining about this so i think it's fine for now.

**Reply by Moonpug-xyz** [reply]
It's really annoying, and a lot of people are complaining. Please remove it or allow users to set their own className.

**Comment by 0xbuidler** [comment]
This is linked to this issue: https://github.com/rainbow-me/rainbowkit/issues/1546

@magiziz
